web/src/main/java/org/fao/geonet/monitor/link/LinkInfo.java

```diff
@@ -0,0 +1,91 @@
+package org.fao.geonet.monitor.link;
+
+import org.jdom.Element;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class LinkInfo {
```

Comment:
Ok, I've got a suggestion for re-writing `getStatus()` to more closely reflect what's described in [the design doc](https://github.com/aodn/internal_discussions/wiki/Geonetwork-Layer-Monitor#linkinfo-state-evaluation):

```java
public LinkMonitorService.Status getStatus() {

  if (notEnoughFreshChecks()) {
    return LinkMonitorService.UNKNOWN;
  }

  if (successfulCheckPercentage() < LinkMonitorService.percentWorkingThreshold) {
    return LinkMonitorService.FAILED;
  }

  return LinkMonitorService.WORKING;
}
```

Maybe this is not quite right - the current implementation has more `return` points than this, but IMO, it'd be really nice if the code reflected the fundamental checking algorithm more closely (with implementation details hidden down a level).  Because... the doc will die, the code will live on.